deps: update V8 to 6.9

targos opened the pull request:
Let's start a new iteration!

ETA: Sept 4th, 2018

targos commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/16013/
V8 CI: https://ci.nodejs.org/job/node-test-commit-v8-linux/1544/

targos commented:
I'm on holiday. Feel free to push anything to my branch

targos commented:
@cjihrig Your patch doesn't apply cleanly. I tried to make your changes by hand but the `test-postmortem-metadata` still fails.

refack commented:
@targos @hashseed Why are these necessary?
- build: update to renamed inspector scripts b3e924b
- tools: update inspector_protocol 45a04fb

I thought the motivation for decoupling `inspector_protocol` from V8 was so that we won't have to update it concurrently with V8?

targos commented:
@refack decoupling inspector_protocol from V8 happened after I first opened this PR so https://github.com/nodejs/node/commit/b3e924b1f01cef73e9346bc29e881c5c49b93315 was already done. I assumed it wouldn't be a bad thing to update the tool but I can revert both commits if needed.

cjihrig commented:
@targos, not sure what happened, but please give https://github.com/cjihrig/node-1/commit/594f6b2df50b81ca605cd6f74a7f4d54b54aa28c a shot, it should work.

targos commented:
@cjihrig Thanks, looks good!

targos commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/16782/
V8: https://ci.nodejs.org/job/node-test-commit-v8-linux/1633/

targos commented:
@nodejs/platform-smartos `v8ustack.d` still needs a fix ;)

```
15:00:52 dtrace: failed to compile script src/v8ustack.d: line 402: failed to resolve V8DBG_CLASS_SHAREDFUNCTIONINFO__FUNCTION_IDENTIFIER__OBJECT: Unknown variable name
15:00:52 node_dtrace_ustack.target.mk:26: recipe for target '/home/iojs/build/workspace/node-test-commit-smartos/nodes/smartos17-64/out/Release/obj.target/node_lib/src/node_dtrace_ustack.o' failed
```

cjihrig commented:
I'll take care of it today.

targos commented:
@nodejs/platform-freebsd Anyone can help with the compilation failure?

https://ci.nodejs.org/job/node-test-commit-freebsd/20036/nodes=freebsd10-64/console

<details>

```
14:45:44 In file included from ../deps/v8/src/asmjs/asm-parser.cc:5:
14:45:44 In file included from ../deps/v8/src/asmjs/asm-parser.h:12:
14:45:44 In file included from ../deps/v8/src/asmjs/asm-types.h:13:
14:45:44 In file included from ../deps/v8/src/zone/zone-containers.h:20:
14:45:44 ../deps/v8/src/zone/zone-allocator.h:57:5: error: no matching function for call to 'operator new'
14:45:44     new (v_p) U(std::forward<Args>(args)...);
14:45:44     ^~~
14:45:44 /usr/include/c++/v1/memory:1572:18: note: in instantiation of function template specialization 'v8::internal::ZoneAllocator<std::__1::__hash_node<std::__1::__hash_value_type<v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int>, void *> >::construct<const v8::internal::Signature<v8::internal::wasm::ValueType>, const v8::internal::Signature<v8::internal::wasm::ValueType> &>' requested here
14:45:44             {__a.construct(__p, _VSTD::forward<_Args>(__args)...);}
14:45:44                  ^
14:45:44 /usr/include/c++/v1/memory:1453:14: note: in instantiation of function template specialization 'std::__1::allocator_traits<v8::internal::ZoneAllocator<std::__1::__hash_node<std::__1::__hash_value_type<v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int>, void *> > >::__construct<const v8::internal::Signature<v8::internal::wasm::ValueType>, const v8::internal::Signature<v8::internal::wasm::ValueType> &>' requested here
14:45:44             {__construct(__has_construct<allocator_type, _Tp*, _Args...>(),
14:45:44              ^
14:45:44 /usr/include/c++/v1/unordered_map:1340:20: note: in instantiation of function template specialization 'std::__1::allocator_traits<v8::internal::ZoneAllocator<std::__1::__hash_node<std::__1::__hash_value_type<v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int>, void *> > >::construct<const v8::internal::Signature<v8::internal::wasm::ValueType>, const v8::internal::Signature<v8::internal::wasm::ValueType> &>' requested here
14:45:44     __node_traits::construct(__na, _VSTD::addressof(__h->__value_.__cc.first), __k);
14:45:44                    ^
14:45:44 /usr/include/c++/v1/unordered_map:1365:25: note: in instantiation of member function 'std::__1::unordered_map<v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int, v8::base::hash<v8::internal::Signature<v8::internal::wasm::ValueType> >, std::__1::equal_to<v8::internal::Signature<v8::internal::wasm::ValueType> >, v8::internal::ZoneAllocator<std::__1::pair<const v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int> > >::__construct_node_with_key' requested here
14:45:44     __node_holder __h = __construct_node_with_key(__k);
14:45:44                         ^
14:45:44 ../deps/v8/src/asmjs/asm-parser.cc:2220:35: note: in instantiation of member function 'std::__1::unordered_map<v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int, v8::base::hash<v8::internal::Signature<v8::internal::wasm::ValueType> >, std::__1::equal_to<v8::internal::Signature<v8::internal::wasm::ValueType> >, v8::internal::ZoneAllocator<std::__1::pair<const v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int> > >::operator[]' requested here
14:45:44       function_info->import->cache[*sig] = index;
14:45:44                                   ^
14:45:44 ../deps/v8/src/zone/zone.h:124:9: note: candidate function not viable: cannot convert argument of incomplete type 'void *' to 'v8::internal::Zone *'
14:45:44   void* operator new(size_t size, Zone* zone) { return zone->New(size); }
14:45:44         ^
14:45:44 1 error generated.
14:45:44 deps/v8/gypfiles/v8_base.target.mk:632: recipe for target '/usr/home/iojs/build/workspace/node-test-commit-freebsd/nodes/freebsd10-64/out/Release/obj.target/v8_base/deps/v8/src/asmjs/asm-parser.o' failed
```

</details>

cjihrig commented:
@targos here is the commit to fix the SmartOS build: https://github.com/cjihrig/node-1/commit/ea184f63c0501d50a5df56ab84c44ed3d2ec0dbe

targos commented:
Thanks again!
CI: https://ci.nodejs.org/view/All/job/node-test-commit-smartos/19858/

targos commented:
@nodejs/platform-freebsd Anyone can help with the compilation failure?

https://ci.nodejs.org/job/node-test-commit-freebsd/20036/nodes=freebsd10-64/console

<details>

```
14:45:44 In file included from ../deps/v8/src/asmjs/asm-parser.cc:5:
14:45:44 In file included from ../deps/v8/src/asmjs/asm-parser.h:12:
14:45:44 In file included from ../deps/v8/src/asmjs/asm-types.h:13:
14:45:44 In file included from ../deps/v8/src/zone/zone-containers.h:20:
14:45:44 ../deps/v8/src/zone/zone-allocator.h:57:5: error: no matching function for call to 'operator new'
14:45:44     new (v_p) U(std::forward<Args>(args)...);
14:45:44     ^~~
14:45:44 /usr/include/c++/v1/memory:1572:18: note: in instantiation of function template specialization 'v8::internal::ZoneAllocator<std::__1::__hash_node<std::__1::__hash_value_type<v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int>, void *> >::construct<const v8::internal::Signature<v8::internal::wasm::ValueType>, const v8::internal::Signature<v8::internal::wasm::ValueType> &>' requested here
14:45:44             {__a.construct(__p, _VSTD::forward<_Args>(__args)...);}
14:45:44                  ^
14:45:44 /usr/include/c++/v1/memory:1453:14: note: in instantiation of function template specialization 'std::__1::allocator_traits<v8::internal::ZoneAllocator<std::__1::__hash_node<std::__1::__hash_value_type<v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int>, void *> > >::__construct<const v8::internal::Signature<v8::internal::wasm::ValueType>, const v8::internal::Signature<v8::internal::wasm::ValueType> &>' requested here
14:45:44             {__construct(__has_construct<allocator_type, _Tp*, _Args...>(),
14:45:44              ^
14:45:44 /usr/include/c++/v1/unordered_map:1340:20: note: in instantiation of function template specialization 'std::__1::allocator_traits<v8::internal::ZoneAllocator<std::__1::__hash_node<std::__1::__hash_value_type<v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int>, void *> > >::construct<const v8::internal::Signature<v8::internal::wasm::ValueType>, const v8::internal::Signature<v8::internal::wasm::ValueType> &>' requested here
14:45:44     __node_traits::construct(__na, _VSTD::addressof(__h->__value_.__cc.first), __k);
14:45:44                    ^
14:45:44 /usr/include/c++/v1/unordered_map:1365:25: note: in instantiation of member function 'std::__1::unordered_map<v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int, v8::base::hash<v8::internal::Signature<v8::internal::wasm::ValueType> >, std::__1::equal_to<v8::internal::Signature<v8::internal::wasm::ValueType> >, v8::internal::ZoneAllocator<std::__1::pair<const v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int> > >::__construct_node_with_key' requested here
14:45:44     __node_holder __h = __construct_node_with_key(__k);
14:45:44                         ^
14:45:44 ../deps/v8/src/asmjs/asm-parser.cc:2220:35: note: in instantiation of member function 'std::__1::unordered_map<v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int, v8::base::hash<v8::internal::Signature<v8::internal::wasm::ValueType> >, std::__1::equal_to<v8::internal::Signature<v8::internal::wasm::ValueType> >, v8::internal::ZoneAllocator<std::__1::pair<const v8::internal::Signature<v8::internal::wasm::ValueType>, unsigned int> > >::operator[]' requested here
14:45:44       function_info->import->cache[*sig] = index;
14:45:44                                   ^
14:45:44 ../deps/v8/src/zone/zone.h:124:9: note: candidate function not viable: cannot convert argument of incomplete type 'void *' to 'v8::internal::Zone *'
14:45:44   void* operator new(size_t size, Zone* zone) { return zone->New(size); }
14:45:44         ^
14:45:44 1 error generated.
14:45:44 deps/v8/gypfiles/v8_base.target.mk:632: recipe for target '/usr/home/iojs/build/workspace/node-test-commit-freebsd/nodes/freebsd10-64/out/Release/obj.target/v8_base/deps/v8/src/asmjs/asm-parser.o' failed
```

</details>

This is the only thing blocking this update.

targos commented:
My proposition to fix this issue: https://github.com/nodejs/node/pull/22617

targos commented:
- FreeBSD 10 support: dropped.
- Chrome 69: released
- This PR: rebased and updated

@nodejs/v8-update @nodejs/tsc @nodejs/intl @nodejs/diagnostics @nodejs/post-mortem  please review :)

A few notes:
- https://github.com/nodejs/node/pull/21983/commits/ff6ad67db93a8987c260e462a0b8c5c6135fcb91 is a semver-major change to the public API
- https://github.com/nodejs/node/pull/21983/commits/3f1250ce809b28901d2e19c24610f8b3667544df adds a new public method to the API
- https://github.com/nodejs/node/pull/21983/commits/7825df0a73454757997e8780c59a3e5c4a33526a changes how ICU is built
- https://github.com/nodejs/node/pull/21983/commits/5b1bfb58fce5a8d318abcd9c9c635d8ed2cfd3d6 removes a file from the release

refack commented:
> I assumed it wouldn't be a bad thing to update the tool but I can revert both commits if needed.

IMHO the `inspector_protocol` commits (b3e924b 45a04fb) could be pulled out. As of https://github.com/nodejs/node/pull/21975 it's an independant tool. And in https://github.com/nodejs/node/pull/22680 I propose a way to simplify node's use of it.

targos commented:
@refack done!

New CI: https://ci.nodejs.org/job/node-test-pull-request/17024/

bricss commented:
Is it will get into Node 10 or only in future Node 11?

targos commented:
CI is green :)

But this still [needs reviews](https://github.com/nodejs/node/pull/21983#issuecomment-418502870).

targos commented:
Rebased again. @nodejs/tsc, as this is semver-major, I need LGTMs from you 🙏

targos commented:
Thank you Matteo!

Landed in 12ed7c94e5...dbf72030b7

addaleax commented:
@targos Do we want to investigate backporting to Node 10? This comes with semver-major changes on *Node’s* side as well, forced by V8 changes, so it would definitely be tricky.
